lxd/storage/drivers/driver_btrfs_utils.go

```diff
@@ -96,18 +96,25 @@ func (d *btrfs) isSubvolume(path string) bool {
 }
 
 func (d *btrfs) getSubvolumes(path string) ([]string, error) {
-	poolMountPath := GetPoolMountPath(d.name)
-	if !strings.HasPrefix(path, poolMountPath+"/") {
-		return nil, fmt.Errorf("%q is outside pool mount path %q", path, poolMountPath)
-	}
-
-	path = strings.TrimPrefix(path, poolMountPath+"/")
-
 	// Make sure the path has a trailing slash.
 	if !strings.HasSuffix(path, "/") {
 		path = path + "/"
 	}
 
+	poolMountPath := GetPoolMountPath(d.name)
```

Comment:
thanks for taking a look at this.

I'm a bit concerned this function is starting to get quite complex to figure out what path to be using.
Is this perhaps an indication that the definition of `path` for this function isn't clear or its being (mis)used in some places?
Should we be doing additional transform outside of the function to path in pool-level situations?

Reply:
I don't find this too complex, and the definition of `path` is also clear. Switching to `btrfs subvolume` just caused a regression which this PR attempts to fix. We should just handle this special case, and leave the rest as it is.

Reply:
I'm wondering why the `path` provided to the function isn't always a valid btrfs subvolume?

Reply:
I've reverted the changes on `getSubvolumes`, and added a new function `hasSubvolumes` as that's what `Create` actually cares about.